### DISCUSSION TITLE
Using Himalaya with manual proxy

### DISCUSSION THREAD

**Discussion by DhruvaSambrani** [pimalaya/himalaya#263] [Q&A] [answered]
My institute has a manual proxy setting that must be set for applications to connect to the internet. This has been a recent development, and all cli email clients seem to stop connecting to the email servers.

For neomutt, there was a closed issue for no one being interested in implementing the feature, and I have opened an issue for aerc. 

Himalaya too doesn't seem to connect to email servers, despite having set http_proxy and https_proxy settings in /etc/environment. All other applications, like curl, work as expected. So, is this a lacking feature in Himalaya (and other clients) or does the NetAdmin have to open some port for email clients to connect?

**Comment by soywod** [comment]
It is a very interesting question. I don't know enough to give an answer straight, but I will dig and let you know. If other tools work as expected with `http_proxy` and/or `https_proxy`, then I would say it is a himalaya bug.

**Reply by DhruvaSambrani** [reply]
Is there anything more I can try? The netadmin also received some emails about Thunderbird not working, but Thunderbird has an explicit proxy setting. This suggests that IMAP probably needs some port open. I will revert back if there is any change.

That said, should Himalaya have a config setting for the same? I'm not entirely sure, but I think people setup personal proxies to anonymize their connections, and may want their email client to also use the proxy.

**Reply by soywod** [reply]
I'm not convinced about having a dedicated config for proxies. Instead, I prefer to rely on standards like `http(s)_proxy`. Which kind of proxy do you have? SOCKS5?

**Reply by DhruvaSambrani** [reply]
I actually have no idea what proxy is being used, just that there is a proxy. If its of any heuristic help, the proxy port is 3128.

**Reply by soywod** [reply]
Do you have the URL of the proxy (if it is a sensitive data, replace sensible parts by fake data)? The shape of the URL can give information about the type of the proxy.

**Reply by DhruvaSambrani** [reply]
The URL is http://172.16.x.xxx:3128 which looks like just an internal IP on the institute LAN.

**Reply by soywod** [reply]
So I guess it is a regular HTTP proxy. I will think about how to implement this. Thanks for the idea!

**Reply by soywod** [reply, marked as answer]
https://github.com/soywod/himalaya/issues/264
